### PR TITLE
add encrypted support

### PR DESCRIPTION
**Pull request by alexcos20**
## Description

Adds encrypted ddo support to aquarius:
 - add encrypt endpoint
 - allow events_monitor to reject a ddo if it's not encrypted and env ONLY_ENCRYPTED_DDO is set to 1

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: aquarius/app/assets.py [hunk 1]
```diff
@@ -492,6 +494,46 @@ def validate_remote():
...
+        description: data
+        schema:
+          type: object
+    responses:
+      200:
+        description: successfully request.
+      400:
+        description: Invalid format
+      500:
+        description: Error
```
Comment (ssallam) [inline aquarius/app/assets.py:525]: are these error codes handled in the implementation below?

File: tests/test_api.py [hunk 2]
```diff
@@ -238,3 +238,17 @@ def test_get_assets_names(client, events_object):
...
+        test_account1, get_web3(), "encrypt_test", json_dict_no_valid_metadata
+    )
+    ddo_string = json.dumps(dict(ddo.items()))
+    compressed_ddo = lzma.compress(Web3.toBytes(text=ddo_string))
+    _response = client.post(
+        base_ddo_url + "/encrypt",
+        data=compressed_ddo,
+        content_type="application/octet-stream",
+    )
+    assert _response.status_code == 200
```
Comment (ssallam) [inline tests/test_api.py:254]: would be nice to test decrypting, and also publishing in the Metadata contract and verifying that the events_monitor is able to decrypt and decompress successfully.

File: aquarius/events/events_monitor.py [hunk 3]
```diff
@@ -579,6 +579,9 @@ def decode_ddo(self, rawddo, flags):
                 logger.error(f"Failed to decompress: {str(err)}")
 
         logger.debug(f"After unpack rawddo:{rawddo}")
+        if self._only_encrypted_ddo and (not check_flags & 2):
+            logger.error("This aquarius can cache only encrypted ddos")
+            return None
```
Comment (ssallam) [inline aquarius/events/events_monitor.py:584]: Is this check better before decrypting the data?

File: tests/test_api.py (hunk 2 above)
Comment (alexcos20) [reply tests/test_api.py:254]: agreed. I will update the tests

**Review by ssallam — APPROVED** [review]
LGTM

**Pull request merged**